new PR for changes to prior blog posts and new frameworks post (#923)
cmu-delphi/www-main: content/blog/2024-01-30-flash-framework.Rmd

<center>
![**Fig 2.** FlaSH's Human-in-the-loop-Framework](/blog/2024-01-30-flash-framework/image2.png)</center>

Combining the methods, engineering, and quality assurance perspectives for event detection from large-scale public health streams, we developed a human-in-the-loop approach to address some fundamental challenges in existing alerting systems. Namely, we rank Delphi's recently updated data by how much it warrants reviewer attention. Reviewers can then decide how many data points they want to investigate, which prioritizes a reviewer's time and does not expect them to review a certain number of alerts flagged via some arbitrary threshold. 

Here's a quick illustration of why the ranking approach works. The most limited resource in this setting is an expert reviewer's time and attention. We want to show this reviewer as many potential events as possible so that they can interpret the data with their known context. 

<center>
![**Fig 3.** Illustration of Ranking vs. Alerts for large data streams](/blog/2024-01-30-flash-framework/image1.png)</center>

> Now imagine each data point per signal was given an outlier score, and there is an arbitrary threshold where scores higher than that threshold are considered outliers. If we develop granular enough rankings, we can directly compare outlier points between different signals without arbitrary thresholds. Then, our data reviewer can focus on data points in their order of importance and stop the review at any time. 

Reviewer: beautiful!

> Developing this framework required new methods, visualizations, and evaluations. Over the past few years, our goal as part of the [FlaSH team](https://delphi.cmu.edu/blog/2024/01/01/introducing-the-flash-group/) was to implement and iterate on this framework. We're excited to share insights from these perspectives as we've moved FlaSH from a sandboxed research project to a deployed part of Delphi's systems that helps our users. For the next post in this series, I'll describe some of the novel methods we use to rank events in large volumes of public health data. Then, Richa and Nolan will share their thoughts on moving these research prototypes into a production environment. Third, Tina and Catalina will describe the data visualization and quality assurance processes and considerations they encountered while facilitating event detection investigation. Beyond that, we'll share updates on how we work towards tightening the human-in-the-loop process!

Reviewer: Insights from Roni: The link to the other blog post is broken (the title of the linked article changed since this one was drafted, plus we should be able to use a relative URI reference) and we should use people's full names (i don't know Catalina's).
```suggestion
Developing this framework required new methods, visualizations, and evaluations. Over the past few years, our goal as part of the [FlaSH team](/blog/2024/01/01/introducing-flash-part-1-meet-the-team/) was to implement and iterate on this framework. We're excited to share insights from these perspectives as we've moved FlaSH from a sandboxed research project to a deployed part of Delphi's systems that helps our users. For the next post in this series, I'll describe some of the novel methods we use to rank events in large volumes of public health data. Then, Richa Gadgil and Nolan Gormley will share their thoughts on moving these research prototypes into a production environment. Third, Tina Townes and Catalina ____ will describe the data visualization and quality assurance processes and considerations they encountered while facilitating event detection investigation. Beyond that, we'll share updates on how we work towards tightening the human-in-the-loop process!
```

---
title: Alerting Systems are Incompatible with Modern Public Health Data Monitoring 
author: Ananya Joshi
date: 2024-01-01
tags:
  - flash
authors:
  - ajoshi
heroImage: blog_alerts_hero.png
heroImageThumb: blog_alerts_thumb.png
output:
  blogdown::html_page:
    toc: true
---

> Insights from public health data can keep communities safe. However, identifying these insights in large volumes of modern public health data can be laborious^1. As a result, over the past few decades, public health agencies have built monitoring systems, like [ESSENCE](https://www.cdc.gov/nssp/new-users.html) (CDC), [EIOS](https://www.who.int/initiatives/eios) (WHO), and [DHIS2](https://dhis2.org/) (WHO), where users can set custom statistical alerts and then investigate these alerts using data visualizations^2. These alerting systems largely follow the following formula^3 as shown in Fig 1.:

Reviewer: The footnote links are still not showing up properly...  It looks like this should work (and then delete the link targets at the bottom of this file).
```suggestion
Insights from public health data can keep communities safe. However, identifying these insights in large volumes of modern public health data can be laborious^[Rosen, George. A history of public health. JHU Press, 2015.]. As a result, over the past few decades, public health agencies have built monitoring systems, like [ESSENCE](https://www.cdc.gov/nssp/new-users.html) (CDC), [EIOS](https://www.who.int/initiatives/eios) (WHO), and [DHIS2](https://dhis2.org/) (WHO), where users can set custom statistical alerts and then investigate these alerts using data visualizations^[Chen, Hsinchun, Daniel Zeng, and Ping Yan. Infectious disease informatics: syndromic surveillance for public health and biodefense. Vol. 21. New York: Springer, 2010.]. These alerting systems largely follow the following formula^[Murphy, Sean Patrick, and Howard Burkom. "Recombinant temporal aberration detection algorithms for enhanced biosurveillance." Journal of the American Medical Informatics Association 15.1 (2008): 77-86.] as shown in Fig 1.:
```

> Insights from public health data can keep communities safe. However, identifying these insights in large volumes of modern public health data can be laborious <sup>[1](#link1)</sup>. As a result, over the past few decades, public health agencies have built monitoring systems, like [ESSENCE](https://www.cdc.gov/nssp/new-users.html) (CDC), [EIOS](https://www.who.int/initiatives/eios) (WHO), and [DHIS2](https://dhis2.org/) (WHO), where users can set custom statistical alerts and then investigate these alerts using data visualizations <sup>[2](#link2)</sup>. These alerting systems largely follow the following formula <sup>[3](#link3)</sup> as shown in Fig 1.:

Reviewer: In the netlify preview, there is weird spacing after each of the links (including the footnote references) in this line/paragraph.  I am not sure whats causing it...  Are there tabs hidden in here instead of spaces?  Is it related to your usage of the `<sup>` tags?  Maybe try [this carat notation](https://www.math.mcgill.ca/yyang/regression/RMarkdown/example.html#:~:text=To%20indicate%20a%20superscript%2C%20use,need%20to%20delimit%20with%20braces.) instead?

Author: I think it should be taken care of now - seems like a sup issue!

<center>
![**Fig 1** Standard Approach for Alerting Systems](/blog/2024-01-30-flash-framework/image3.png)</center>

> 1. Predict a value that you expect the data to look like 

Reviewer: nit:
```suggestion
1. Predict a value that you expect the data to have
```

2. Calculate the difference between your observed and predicted values 

> 3. Contextualize the difference (e.g., to account for the variability in the data)

Reviewer: nit:
```suggestion
3. Contextualize the difference (e.g., to account for variability in the data)
```

4. Send alerts when this value passes a threshold

> Using this formula, if we assume that our data follows a bell curve, we can calculate the difference between our prediction (which might be the average of the data) and what we observe and divide by the standard deviation of the data. If this value exceeds a specific threshold of 3, then the probability that we would observe that data point in that bell curve would be less than 1%, which means that it is quite surprising. By setting this threshold high, only a small percentage of points are identified as outliers and result in alerts.

Reviewer: nit(s) -- take these suggestions with a grain of salt, im not sure if im capturing all of your intent.  i am also not clear on where the "1%" comes from or how it relates to the threshold of 3.
```suggestion
Using this formula, if we assume that our data follows the bell curve of a normal distribution, we can calculate the difference between our prediction (which might be the average of the data) and what we observe, and divide by the standard deviation. If this value exceeds a specific threshold (here we use 3), then the probability that we would observe that data point in that bell curve would be less than 1%, which means that it is quite surprising. By setting this threshold high, only a small percentage of points are identified as outliers and thus result in alerts.
```

Author: Ah ! so for the normal distribution specifically, 99% of the data is contained within 3 standard deviations, which means data beyond those 3 deviations (this is a continuous distribution which can take any value between +-inf) is very unlikely (and only <1% of the distribution takes on those values). Let me know if that doesn't make sense!

This was the same intuition we used while initially designing Delphi's data monitoring system at the pandemic's start. This monitoring system was particularly important because Delphi's stakeholders wanted to be aware of any events in the data - like changes in data quality or disease dynamics. First, we split these events into data validation (e.g., negative cases) and events that needed a human-facing statistical alert (using the above formula) so that an expert could ultimately interpret and classify the event. 

> However, we quickly realized this formula did not work with Delphi's large data volume. To illustrate, creating alerts for 1% of the data over millions of data points, like those analyzed by Delphi, results in 10,000+ alerts - far too many for us at Delphi to inspect and investigate! Standard correction approaches were also incompatible with public health data. Our first attempt was reducing the percentage of alerts (e.g., thresholds that would identify from 1% of the data as alerts to 0.01%). However, because modeling public health data is difficult, defining thresholds (e.g., 1 in 10000) when data streams only have a few hundred historical data points can be somewhat arbitrary. Additionally, when applying the same standard threshold-generating function across all the data (adaptive thresholds), we missed some important events from signals that were not well-calibrated to the function. Tuning these parameters by hand across hundreds of signals was tedious, and Delphi turned off the outlier alerting system. 

Reviewer: nit:
```suggestion
However, we quickly realized this formula did not work with Delphi's large data volume. To illustrate: creating alerts for 1% of the data over millions of data points, like those analyzed by Delphi, results in 10,000+ alerts - far too many for us at Delphi to inspect and investigate! Standard correction approaches were also incompatible with public health data. Our first attempt was reducing the percentage of alerts (e.g., changing thresholds that would identify from 1% of the data as alerts to 0.01%). However, because modeling public health data is difficult, defining thresholds (e.g., 1 in 10000) when data streams only have a few hundred historical data points can be somewhat arbitrary. Additionally, when applying the same standard threshold-generating function across all the data (adaptive thresholds), we missed some important events from signals that were not well-calibrated to the function. Tuning these parameters by hand across hundreds of signals was tedious, and Delphi turned off the outlier alerting system. 
```

> The historical alerting framework cannot keep up with monitoring the increasingly large volumes of heterogeneous public health data that organizations like Delphi process. Delphi has increased its volume by 1000x in the past three years alone. And, although Delphi processes data from several geographical regions, even local public health agencies who use systems (that focus on one geography) under the existing alerting framework have challenged its usefulness (e.g. "[What Can You Really Do with 35,000 Statistical Alerts a Week Anyways?](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6606136/)" (2019)). We anticipate this failure mode for alerting systems will become more prevalent as data volumes grow. 

Reviewer: nit?
```suggestion
The historical alerting framework cannot keep up with monitoring the increasingly large volumes of heterogeneous public health data that organizations like Delphi process. Delphi has increased its volume by 1000x in the past three years alone. And, while some organizations struggle to processes data from numerous geographical regions, even local public health agencies (focusing on smaller geographies) question the usefulness of the existing alerting framework (e.g. "[What Can You Really Do with 35,000 Statistical Alerts a Week Anyways?](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6606136/)" (2019)). We anticipate this failure mode for alerting systems will become more prevalent as data volumes grow. 
```

> ### Bridging the gap with FlaSH:

Reviewer: colon is not necessary because this is a header
```suggestion
### Bridging the gap with FlaSH
```

> Identifying and communicating noteworthy events from public health data is a critical step for future pandemic prevention <sup>[4](#link4)</sup>. But how should we redesign our event detection process? 

Reviewer: This superscript does not suffer the same spacing strangeness as in the first paragraph.  Very weird!

> After reflecting on our prior system, we realized the reason why existing monitoring fail, best summarized by the following quote: 

Reviewer: nit:
```suggestion
After reflecting on our prior system, we realized the reason why existing monitoring fails is best summarized by the following quote: 
```

<blockquote>
"The current disconnect among algorithm developers, implementers, and [experts] … foster[s] distrust in statistical monitoring and in biosurveillance itself."  - Galit Shmeuli & Howard Burkom <sup>[5](#link5)</sup>
</blockquote>
    
Interestingly, through our prior system, I had the opportunity to experience all three roles, from developing the outlier detection methods, implementing and deploying the methods over Delphi's data, and then analyzing the outputs of the methods. These three roles came with different constraints and priorities. These experiences set the foundation for devising an interdisciplinary approach. 

<center>
![**Fig 2.** FlaSH's Human-in-the-loop-Framework](/blog/2024-01-30-flash-framework/image2.png)</center>

> Combining the methods, engineering, and quality assurance perspectives for event detection from large-scale public health streams, we developed a human-in-the-loop approach to address some fundamental challenges in existing alerting systems. Namely, we rank Delphi's recently updated data by how much it warrants reviewer attention. Reviewers can then decide how many data points they want to investigate, which prioritizes a reviewer's time and does not expect them to review a certain number of alerts flagged via a somewhat arbitrary threshold. 

Reviewer: nit?
```suggestion
Combining the methods, engineering, and quality assurance perspectives for event detection from large-scale public health streams, we developed a human-in-the-loop approach to address some fundamental challenges in existing alerting systems. Namely, we rank Delphi's recently updated data by how much it warrants reviewer attention. Reviewers can then decide how many data points they want to investigate, which prioritizes a reviewer's time and does not expect them to review a certain number of alerts flagged via some arbitrary threshold. 
```

> Here's a quick illustration of why the ranking approach works. The most limited resource in this setting is an expert reviewer's time and attention. We want to show this reviewer as many possible events as possible so that they can interpret their data with their known context. 

Reviewer: ```suggestion
Here's a quick illustration of why the ranking approach works. The most limited resource in this setting is an expert reviewer's time and attention. We want to show this reviewer as many potential events as possible so that they can interpret the data with their known context. 
```

<center>
![**Fig 3.** Illustration of Ranking vs. Alerts for large data streams](/blog/2024-01-30-flash-framework/image1.png)</center>

> Now imagine each data point per signal was given an outlier score, and there is an arbitrary binary threshold over all the data. If we develop granular enough rankings, we can directly compare outlier points between different signals without arbitrary thresholds. Then, a data reviewer can then focus on data points in their order of importance and stop the review at any time. 

Reviewer: nits (also im not sure what you mean  by "arbitrary binary threshold") :
```suggestion
Now imagine each data point was given an outlier score, and there is an arbitrary binary threshold over all the data. If we develop granular enough rankings, we can directly compare outlier points between different signals more objectively. Then, our data reviewer can focus on data points in their order of importance and stop the review at any time. 
```

Author: This is supposed to show that if you set a threshold like 0.9 above which everything is an outlier and below which everything is not an outlier, that's a bit of an arbitrary number  (what about 0.89? why not 0.91?). So if you use rankings, you can combine these scores from different signals in a meaningful way because otherwise, there might just be a lot of things that are classified as outliers even if there is variation in those scores.

Author: Let me know if you think there's a clearer way to share this - I was hoping the picture would make it more clear, but if not this is the most important part of the blog, so maybe we can iterate on how to make it more understandable!

Author: @melange396 
Just wanted to double check on this part - does it make more sense now?

Reviewer: it makes sense.  it was the word "binary" that threw me off; you mention thresholds a number of times before this, but this is the only time you call it a "binary threshold".

> Developing this framework required new methods, visualizations, and evaluations. Over the past few years, our goal as part of the [FlaSH team](https://delphi.cmu.edu/blog/2024/01/01/introducing-the-flash-group/) was to implement and iterate on this framework. We're excited to share insights from these perspectives as we've moved FlaSH from a sandboxed research project to a deployed part of Delphi's systems that helps our users. For the next post in this series, I'll describe some of the novel methods we use to rank events in large volumes of public health data. Then, Richa and Nolan will share their perspectives on moving these research prototypes into a production environment. Third, Tina and Catalina will describe the data visualization and quality assurance perspectives and considerations while facilitating event detection investigation. Beyond that, we'll routinely share updates on how we work towards tightening the human-in-the-loop process!

Reviewer: one more wording nit, plus i would remove "routinely" just in case it doesn't end up truly being routine.
 
```suggestion
Developing this framework required new methods, visualizations, and evaluations. Over the past few years, our goal as part of the [FlaSH team](https://delphi.cmu.edu/blog/2024/01/01/introducing-the-flash-group/) was to implement and iterate on this framework. We're excited to share insights from these perspectives as we've moved FlaSH from a sandboxed research project to a deployed part of Delphi's systems that helps our users. For the next post in this series, I'll describe some of the novel methods we use to rank events in large volumes of public health data. Then, Richa and Nolan will share their perspectives on moving these research prototypes into a production environment. Third, Tina and Catalina will describe the data visualization and quality assurance perspectives and considerations they encountered while facilitating event detection investigation. Beyond that, we'll share updates on how we work towards tightening the human-in-the-loop process!
```

> Developing this framework required new methods, visualizations, and evaluations. Over the past few years, our goal as part of the [FlaSH team](https://delphi.cmu.edu/blog/2024/01/01/introducing-the-flash-group/) was to implement and iterate on this framework. We're excited to share insights from these perspectives as we've moved FlaSH from a sandboxed research project to a deployed part of Delphi's systems to help our data users. For the next post in this series, I'll describe some of the novel methods we use to rank events in large volumes of public health data. Then, Richa and Nolan will share their perspectives on moving these research prototypes into a production environment. Third, Tina and Catalina will describe the data visualization and quality assurance perspectives and considerations while facilitating event detection investigation. Beyond that, we'll routinely share updates on how we work towards tightening the human-in-the-loop process!

Reviewer: nit:
```suggestion
Developing this framework required new methods, visualizations, and evaluations. Over the past few years, our goal as part of the [FlaSH team](https://delphi.cmu.edu/blog/2024/01/01/introducing-the-flash-group/) was to implement and iterate on this framework. We're excited to share insights from these perspectives as we've moved FlaSH from a sandboxed research project to a deployed part of Delphi's systems that helps our users. For the next post in this series, I'll describe some of the novel methods we use to rank events in large volumes of public health data. Then, Richa and Nolan will share their perspectives on moving these research prototypes into a production environment. Third, Tina and Catalina will describe the data visualization and quality assurance perspectives and considerations while facilitating event detection investigation. Beyond that, we'll routinely share updates on how we work towards tightening the human-in-the-loop process!
```

The historical alerting framework cannot keep up with monitoring the increasingly large volumes of heterogeneous public health data that organizations like Delphi process. Delphi has increased its volume by 1000x in the past three years alone. And, although Delphi processes data from several geographical regions, even local public health agencies who use systems (that focus on one geography) under the existing alerting framework have challenged its usefulness (e.g. "[What Can You Really Do with 35,000 Statistical Alerts a Week Anyways?](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6606136/)" (2019)). We anticipate this failure mode for alerting systems will become more prevalent as data volumes grow. 

### Bridging the gap with FlaSH

> Identifying and communicating noteworthy events from public health data is a critical step for future pandemic prevention^4. But how should we redesign our event detection process? 

Reviewer: ```suggestion
Identifying and communicating noteworthy events from public health data is a critical step for future pandemic prevention^[“Rapidly Detecting and Responding to Health Emergencies.” World Health Organization, <https://www.who.int/activities/rapidly-detecting-and-responding-to-health-emergencies>. Accessed 29 Jan. 2024.]. But how should we redesign our event detection process? 
```

After reflecting on our prior system, we realized the reason why existing monitoring fails is best summarized by the following quote: 

<blockquote>

> "The current disconnect among algorithm developers, implementers, and [experts] … foster[s] distrust in statistical monitoring and in biosurveillance itself."  - Galit Shmeuli & Howard Burkom^5

Reviewer: ```suggestion
"The current disconnect among algorithm developers, implementers, and [experts] … foster[s] distrust in statistical monitoring and in biosurveillance itself."  - Galit Shmeuli & Howard Burkom^[Shmueli, Galit, and Howard Burkom. "Statistical challenges facing early outbreak detection in biosurveillance." Technometrics 52.1 (2010): 39-51.]
```

Developing this framework required new methods, visualizations, and evaluations. Over the past few years, our goal as part of the [FlaSH team](https://delphi.cmu.edu/blog/2024/01/01/introducing-the-flash-group/) was to implement and iterate on this framework. We're excited to share insights from these perspectives as we've moved FlaSH from a sandboxed research project to a deployed part of Delphi's systems that helps our users. For the next post in this series, I'll describe some of the novel methods we use to rank events in large volumes of public health data. Then, Richa and Nolan will share their thoughts on moving these research prototypes into a production environment. Third, Tina and Catalina will describe the data visualization and quality assurance processes and considerations they encountered while facilitating event detection investigation. Beyond that, we'll share updates on how we work towards tightening the human-in-the-loop process!

This blog post includes material from my thesis proposal, "Identifying Anomalous Events in Modern
Public Health Data Streams" 

> <a id="#link1">1.</a>  Rosen, George. A history of public health. JHU Press, 2015.  <br>
> <a id="#link2">2.</a>  Chen, Hsinchun, Daniel Zeng, and Ping Yan. Infectious disease informatics: syndromic surveillance for public health and biodefense. Vol. 21. New York: Springer, 2010. <br>
> <a id="#link3">3.</a>  Murphy, Sean Patrick, and Howard Burkom. "Recombinant temporal aberration detection algorithms for enhanced biosurveillance." Journal of the American Medical Informatics Association 15.1 (2008): 77-86. <br>
> <a id="#link4">4.</a> “Rapidly Detecting and Responding to Health Emergencies.” World Health Organization, ]www.who.int/activities/rapidly-detecting-and-responding-to-health-emergencies. Accessed 29 Jan. 2024. <br>
> <a id="#link5">5.</a> Shmueli, Galit, and Howard Burkom. "Statistical challenges facing early outbreak detection in biosurveillance." Technometrics 52.1 (2010): 39-51.
> 

Reviewer: no longer needed, the carat notation for footnotes automatically puts the text at the bottom of the page.
```suggestion

```
